src/managers/ModelManager.cpp

```diff
@@ -86,32 +93,40 @@ void ModelManager::constructFinishedModels(TimeSeriesModelContainer& finishedSeg
     float pmcMeanSize = finishedSegment.pmcMean.getBytesPerValue();
     float swingSize = finishedSegment.swing.getBytesPerValue();
     float gorillaSize = finishedSegment.gorilla.getBytesPerValue();
-    int lastModelledTimestamp = 0;
+    int lastModelledTimestamp;
+    size_t indexToStart = std::min<size_t>(finishedSegment.pmcMean.get_length(), std::min<size_t>(finishedSegment.gorilla.get_length_gorilla(), finishedSegment.swing.getLength()));
 
-    if (pmcMeanSize > swingSize && pmcMeanSize > gorillaSize){
+    if (pmcMeanSize < swingSize && pmcMeanSize < gorillaSize){
         lastModelledTimestamp = finishedSegment.pmcMean.lastTimestamp;
-    } else if (swingSize > pmcMeanSize && swingSize > gorillaSize){
+        indexToStart = finishedSegment.pmcMean.get_length() - indexToStart;
+    } else if (swingSize < pmcMeanSize && swingSize < gorillaSize){
         lastModelledTimestamp = finishedSegment.swing.get_last_timestamp();
+        indexToStart = finishedSegment.swing.getLength() - indexToStart;
     } else {
         lastModelledTimestamp = finishedSegment.gorilla.lastTimestamp;
+        indexToStart = finishedSegment.gorilla.get_length_gorilla() - indexToStart;
     }
-    if (finishedSegment.cachedValues.startTimestamp != NULL){
+    if (finishedSegment.cachedValues.startTimestamp != 0){
         CachedValues innerCache = std::move(finishedSegment.cachedValues);
         // TODO: MAYBE MOVE
-        finishedSegment = TimeSeriesModelContainer(finishedSegment.errorBound, finishedSegment.errorAbsolute, finishedSegment.id);
+        finishedSegment = TimeSeriesModelContainer(finishedSegment.errorBound, finishedSegment.errorAbsolute, finishedSegment.localId, finishedSegment.globalId);
+
         // TODO: get last constructed TS, and parse rest TS to fitTimeSeriesModels
-        int startIndex = 0, endIndex = 0;
-        timestampManager.calcIndexRangeFromTimestamps(lastModelledTimestamp, lastTimestamp, startIndex, endIndex);
-        std::vector<int> stamps;
-        //timestampManager.getTimestampFromIndex()
-        for (auto value : finishedSegment.cachedValues.values){
-            fitTimeSeriesModels(finishedSegment.id, value);
+        std::vector<int> timestamps = timestampManager.getTimestampRangeForColumnsByTimestamp(finishedSegment.globalId, lastModelledTimestamp, lastTimestamp);

```

Comment:
"getTimestampRangeForColumnsByTimestamp", det her variabel navn er halvdelen af den totale max længde for en linje...

Reply:
Lige knap

Reply:
Men get the point, fixed